### PR TITLE
Introduce `.secondary-nav` and handle `.page-content` spacing universally

### LINKED ISSUE #29981: Login UI layout issue, the top marin of the login div missed.
image

### PR DESCRIPTION
**Pull request by silverwind**
Fixes: https://github.com/go-gitea/gitea/issues/29981. Introduce `.secondary-nav` as a universal way for styling and margin adjustments inside `.page-content`.

If the first child of `.page-content` is `.secondary-nav`, we add margin below it, otherwise we add padding to the first child. Notable changes:

- `--color-header-wrapper` is replaced with `--color-secondary-nav-bg`.
- `navbar` class is removed.

### CODE REVIEW

**Comment by silverwind** [comment]
Maybe later refactor to always have the `overflow-menu` and hide that box header.

**Comment by silverwind** [comment]
Maybe it should not use `:has` because PaleMoon [does not have it](https://repo.palemoon.org/MoonchildProductions/UXP/issues/2473) yet. I will check for alternatives.

**Comment by silverwind** [comment]
> Maybe it should not use `:has` because PaleMoon [does not have it](https://repo.palemoon.org/MoonchildProductions/UXP/issues/2473) yet. I will check for alternatives.

done

**Comment by wxiaoguang** [comment]
Is it really right? I guess it only hacky patches one page you know, but there are a lot.

I would say abusing `tw-` helpers to fine-tune everything is wrong. Please use some well-designed and stable framework-level design.

image

**Comment by wxiaoguang** [comment]
And I would suggest to revert #29922 and leave enough time to have a better framework-level design.

**Comment by silverwind** [comment]
It's quite simple. The first element inside `.page-content` sets padding. Here, either the `<overflow-menu>` or the login box is first element and the former currently has a built-in margin, so adding to the box would make a double margin.

If you want it cleaner, I will remove the built-in margin on `<overflow-menu>` and instead add it via `tw-` helpers to all other elements except this one.

And no, I disagree to reverting https://github.com/go-gitea/gitea/pull/29922, it is a important step.

**Comment by wxiaoguang** [comment]
I am sure your hacky patches are not right.

For example, this time, you fixed one page:

image


But there is another page:

image


PLEASE provide a stable framework-level design.

**Comment by silverwind** [comment]
There is no general solution here. Some elements want margin, some don't. Some want different margin even.

**Comment by silverwind** [comment]
So in essence we have two scenarios:

- There is a secondary navbar. The second element needs the margin.
- There is no secondary navbar. The first element needs the margin.

So maybe a general solution is possible but the selector for second element won't be pretty.

**Comment by silverwind** [comment]
@wxiaoguang what to you think about the above suggestion? Is that "framework" enough for you?

**Comment by wxiaoguang** [comment]
> @wxiaoguang what to you think about the above suggestion? Is that "framework" enough for you?

If I understand correctly:

```
nvarbar 
(15 margin)
page-content
```

```
nvarbar 
secondary-navbar
(15 margin)
page-content
```


right? It looks like the same as what Fomantic UI does: `attached segment`. I think it would work.

**Comment by silverwind** [comment]
Yes, like that. On repo page it currently uses 12px, but stuff like that can be overridden with tailwind and maybe we should just generally reduce from 15px to 12px.

**Comment by silverwind** [comment]
Actually on repo the `.header-wrapper` fulfills the role of the secondary navbar and the 12px margin is internal to that, so we can still make it standard margin below.

**Comment by wxiaoguang** [comment]
> Yes, like that. On repo page it currently uses 12px, but stuff like that can be overridden with tailwind and maybe we should just generally reduce from 15px to 12px.

Some general and well-defined classes look good to me.

**Comment by silverwind** [comment]
Ok I guess I will remove this `header-wrapper` class and replace it with `secondary-navbar`. There is also a CSS variable for it that I will change.

**Comment by silverwind** [comment]
Ready again, this should now cover all cases univerally with "framework" level support 😆.

File: web_src/css/base.css [hunk 1]
```diff
@@ -1337,6 +1340,10 @@ overflow-menu .overflow-menu-items .item {
   margin-bottom: 0 !important; /* reset fomantic's margin, because the active menu has special bottom border */
 }
 
+overflow-menu .ui.label {
+  margin-left: 7px !important;
+}
+
```
Comment (silverwind) [inline web_src/css/base.css:1348]: This rule was previously in repo only, but we can make this apply to all such menus.

File: web_src/css/repo/header.css [hunk 2]
```diff
@@ -1,4 +1,8 @@
-.header-wrapper .fork-flag {
+.repository .secondary-nav {
+  padding-top: 12px;
+}
+
+.repository .secondary-nav .fork-flag {
```
Comment (wxiaoguang) [inline web_src/css/repo/header.css:5]: Why this change?

**Review by wxiaoguang — APPROVED** [review]

File: web_src/css/repo/header.css (hunk 2 above)
Comment (silverwind) [reply web_src/css/repo/header.css:5]: Hmm yes we could remove the `.repository` prefix so it's equivalent to before.

**Review by delvh — APPROVED** [review]

File: web_src/css/base.css (hunk 1 above)
Comment (wxiaoguang) [reply web_src/css/base.css:1348]: Could there be a comment to explain why "ui label" needs such "margin-left" ?

File: web_src/css/repo/header.css (hunk 2 above)
Comment (silverwind) [reply web_src/css/repo/header.css:5]: Done in https://github.com/go-gitea/gitea/pull/29982/commits/44585322d49477810d30ea3d289d4d6b706cec2b.

Comment (wxiaoguang) [reply web_src/css/repo/header.css:5]: I mean ... it should remove `.secondary-nav` and keep it as before?

File: web_src/css/base.css (hunk 1 above)
Comment (silverwind) [reply web_src/css/base.css:1348]: It's a space-saving measure. Default is `1em` from `.ui.menu .item > .label:not(.floating)`.

Comment (silverwind) [reply web_src/css/base.css:1348]: Added comment in https://github.com/go-gitea/gitea/pull/29982/commits/fa5167ec4f2c139a25c738defd15e90853a7148e.

File: web_src/css/repo/header.css (hunk 2 above)
Comment (silverwind) [reply web_src/css/repo/header.css:5]: Previously it was `.header-wrapper` and that class is replaced with `.secondary-nav` so it's a 1:1 class replacement in the selector.

File: web_src/css/repo/header.css [hunk 3]
```diff
@@ -1,4 +1,8 @@
-.header-wrapper .fork-flag {
+.repository .secondary-nav {
+  padding-top: 12px;
+}
+
+.secondary-nav .fork-flag {
```
Comment (wxiaoguang) [inline web_src/css/repo/header.css]: Oops, I misread, I think your old code is better ....

```suggestion
.repository .secondary-nav .fork-flag {
```

**Comment by silverwind** [comment]
Leaving open a bit more.
